improved existing content, added technical details for byoti (#22986)
DataDog/documentation: content/en/security/application_security/threats/threat-intelligence.md

title: Threat Intelligence
kind: documentation
further_reading:
- link: "https://docs.datadoghq.com/security/threat_intelligence/"
  tag: "Documentation"
  text: "Threat Intelligence at Datadog"
- link: "/security/application_security/"
  tag: "Documentation"
  text: "Protect against threats with Datadog Application Security Management"
---

## Overview

This topic describes [threat intelligence][1] for Application Security Management (ASM).

> Datadog provides built-in threat intelligence [datasets](https://docs-staging.datadoghq.com/michael_bentley/DOCS-7745/security/threat_intelligence#threat-intelligence-sources) for ASM. This provides additional evidence when acting on security activity and reduces detection thresholds for some business logic detections. 

Reviewer: ```suggestion
Datadog provides built-in threat intelligence [datasets][1] for ASM. This provides additional evidence when acting on security activity and reduces detection thresholds for some business logic detections. 
```

  tag: "Documentation"
  text: "Threat Intelligence at Datadog"
- link: "/security/application_security/"
  tag: "Documentation"
  text: "Protect against threats with Datadog Application Security Management"
---

## Overview

This topic describes [threat intelligence][1] for Application Security Management (ASM).

Datadog provides built-in [threat intelligence][1] datasets for ASM. This provides additional evidence when acting on security activity and reduces detection thresholds for some business logic detections. 

Additionally, ASM supports *bring your own threat intel* (private beta). This functionality enriches detections with business-specific threat intelligence. 

Datadog curates threat intelligence into a standardized list of categories and intents. Intents include _benign_, _suspicious_, and _malicious_. 

Categories of threat intelligence include benign detections such as _corp\_vpn_ and malicious categories like _botnet_proxy_. 

Upstream threat intelligence information is passed through for all threat intelligence sources, with limits based on threat intelligence payload size.

Datadog recommends the following methods for consuming threat intelligence:

> 1. Reducing detection rule thresholds for business logic threats such as credential stuffing. Users can clone the default [Credential Stuffing](https://app.datadoghq.com/security/configuration/asm/rules/view/wnp-zlu-woa) rule and modify it to meet their needs.

Reviewer: ```suggestion
1. Reducing detection rule thresholds for business logic threats such as credential stuffing. Users can clone the default [Credential Stuffing][6] rule and modify it to meet their needs.
```

    @appsec.threat_intel:true 

## Bring your own threat intelligence

<div class="alert alert-info">Bring your own threat intelligence is in private beta.</div>

ASM supports enriching and searching traces with threat intelligence indicators of compromise stored in Datadog reference tables. [Reference Tables][2] allow you to combine metadata with information already in Datadog.

### Storing indicators of compromise in reference tables

Threat intelligence is supported in the CSV format and requires 4 columns.

**CSV Structure**

| field            | data  | description| required | example|
|------------------|-------|----|-----|--|
| ip_address       | text | The primary key for the reference table in the IPv4 dot notation format. | true | 192.0.2.1  |
| additional_data  | json      | Additional data to enrich the trace. | false | `{"ref":"hxxp://example.org"}`

> | category         | text  | The threat intel [category](https://docs-staging.datadoghq.com/michael_bentley/DOCS-7745/security/threat_intelligence#threat-intelligence-categories). This is used by some out of the box detection rules. | true | `residential_proxy` |
> | intention        | text | The threat intel [intent](https://docs-staging.datadoghq.com/michael_bentley/DOCS-7745/security/threat_intelligence#threat-intelligence-intents). This is used by some out of the box detection rules.| true | malicious | |
> | source           | text  | The name of the source and the link to its site, such as your team and your teams wiki. | true| `{"name":"internal_security_team", "url":"https://teamwiki.example.org"}` | | 

Reviewer: ```suggestion
| category         | text  | The threat intel [category][7]. This is used by some out of the box detection rules. | true | `residential_proxy` |
| intention        | text | The threat intel [intent][8]. This is used by some out of the box detection rules.| true | malicious | |
| source           | text  | The name of the source and the link to its site, such as your team and your teams wiki. | true| `{"name":"internal_security_team", "url":"https://teamwiki.example.org"}` | | 
```

The full list of supported categories and intents is available at [Threat Intelligence Facets][3].

<div class="alert alert-info">JSON in a CSV requires double quoting. The following is an example CSV.</div>

```
ip_address,additional_data,category,intention,source
192.0.2.1,"{""ref"":""hxxp://example.org""}",scanner,suspicious,"{""name"":""internal_security_team"", ""url"":""https://teamwiki.example.org""}"
192.0.2.2,"{""ref"":""hxxp://example.org""}",scanner,suspicious,"{""name"":""internal_security_team"", ""url"":""https://teamwiki.example.org""}"
192.0.2.3,"{""ref"":""hxxp://example.org""}",scanner,suspicious,"{""name"":""internal_security_team"", ""url"":""https://teamwiki.example.org""}"
```

### Uploading and enabling your own threat intel

On a new [references table][4] page:

> 1. Name the table. The table name will be referenced in ASM's **Threat Intel** config.

Reviewer: ```suggestion
1. Name the table. The table name is referenced in ASM's **Threat Intel** config.
```

{{< partial name="whats-next/whats-next.html" >}}

> [1]: /security/threat_intelligence

Reviewer: ```suggestion
[1]: /security/threat_intelligence/#threat-intelligence-sources
```

[2]: /integrations/guide/reference-tables
[3]: /security/threat_intelligence/#threat-intelligence-facets
[4]: https://app.datadoghq.com/reference-tables/create
[5]: https://app.datadoghq.com/security/configuration/asm/threat-intel

> [6]: https://app.datadoghq.com/security/configuration/asm/rules/edit/kdb-irk-nua?product=appsec

Reviewer: ```suggestion
[6]: https://app.datadoghq.com/security/configuration/asm/rules/edit/kdb-irk-nua?product=appsec
[7]: /security/threat_intelligence#threat-intelligence-categories
[8]: /security/threat_intelligence#threat-intelligence-intents
```
